src/compile/data/source.ts

```diff
@@ -65,13 +66,13 @@ export class SourceNode extends DataFlowNode {
     this._name = name;
   }
 
-  set parent(parent: DataFlowNode) {
-    throw new Error('Source nodes have to be roots.');
-  }
+  // set parent(parent: DataFlowNode) {
```

Comment:
make sense. :) I was trying to reason my way around the ordering. This was definitely something I wasn't sure about!